src/components/FormElements/ResourceFeePickerWithQuery.tsx

```diff
@@ -103,12 +104,27 @@ export const ResourceFeePickerWithQuery = ({
 
   // Create a sample transaction to simulate to get the min resource fee
   const buildTxToSimulate = () => {
+    setErrorMessage("");
+
     try {
-      const contractDataXDR = getContractDataXDR({
-        contractAddress: operation.params.contract,
-        dataKey: operation.params.key_xdr,
-        durability: operation.params.durability,
-      });
+      let contractDataXDR;
+
+      // restore_footprint operation already has ContractDataXDR in base64
+      if (
+        operation.operation_type === "restore_footprint" &&
+        operation.params.contractDataLedgerKey
```

Comment:
Do we need to make sure that `operation.operation_type === "restore_footprint"` applies to both cases (if and else)?

Reply:
at the moment `operation.params.contractDataLedgerKey` ties to only `restore_footprint`